In [1]:
import thinkstats2
import thinkplot
import nsfg
import math

In [2]:
df = nsfg.ReadFemPreg()

In [3]:
#look @ live births only
live = df[df.outcome == 1]


In [4]:
#totalwgt_lb (wt in lb)
firsts = live[live.birthord == 1]
others = live[live.birthord != 1]

In [5]:
# textbook code for calculating Cohen's d
# (how much difference is there in the means in weighted average standard deviations?)
def CohenEffectSize(group1, group2):
    diff = group1.mean() - group2.mean()
    var1 = group1.var()
    var2 = group2.var()
    n1, n2 = len(group1), len(group2)
    pooled_var = (n1 * var1 + n2 * var2) / (n1 + n2)
    d = diff / math.sqrt(pooled_var)
    return d

In [6]:
#get the respective series and feed them into the Cohen's d function
firstwts = firsts.totalwgt_lb
notfirstwts = others.totalwgt_lb
cohens_d = CohenEffectSize(firstwts, notfirstwts)

In [7]:
print(cohens_d)

-0.088672927072602


* non-first-borns are lighter by -0.089 weighted avg standard deviations
* between very small and small according to research (https://en.wikipedia.org/wiki/Effect_size#Cohen.27s_d)
* thus, birth order does not appear to have a very significant effect, small at best